xds/internal/resolver/serviceconfig.go

```diff
@@ -397,9 +409,24 @@ func (r *xdsResolver) newConfigSelector(su serviceUpdate) (*configSelector, erro
 	return cs, nil
 }
 
+// initializeCluster initializes entries in cs.clusters map, creating entries in
+// r.activeClusters as necessary.  Any created entries will be set to zero as
+// they will be incremented by incRefs.
+func (r *xdsResolver) initializeCluster(clusterName string, cs *configSelector) {
```

Comment:
I think this wants to be a method on `configSelector` instead.  It can access the `xdsResolver` that way, too for adding to `activeClusters`.  You can also pass it a `balancerConfig` to store in the `clusterInfo` as mentioned above.

Reply:
Ah, great idea. These two comments make it much cleaner. Switched.